cis/processor.py

```diff
@@ -32,6 +34,11 @@ def run(self):
         # Determine what stage of processing we are in and call the corresponding functions.
         self.decrytped_profile = json.loads(self._decrypt_profile_packet())
 
+        if self.signature == {} or self.signature is None:
+            self.signature = self.decrytped_profile.get('signature')
+
+        self._verify_signature()
```

Comment:
should we fail here, or at least warn for now if this returns False?

Reply:
We actually will fail it.  I put signature verification in this parent object so it would be accessible in all of the objects that take OperationDelgate as a class.  Basicall if you look down in line 105 this is where that "fail and log behavior" is in code.